# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1296120399 18000
#      Thu Jan 27 04:26:39 2011 -0500
# Node ID 0947d399cff44e2568e707bf5cfb8029560466e2
# Parent  a71c1aa9823e6db3c0c9993286d97f66d1b19495
input.cc (input_event_hook): fix incorrect use of iterator

diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,8 +1,13 @@
+2011-01-27  John W. Eaton  <jwe@octave.org>
+
+	* input.cc (input_event_hook): Fix incorrect use of iterator.
+	Reported by Pascal Dupuis <Pascal.Dupuis@worldonline.be>.
+
 2011-01-27  John W. Eaton  <jwe@octave.org>
 
 	* ov-class.cc (octave_class::reconstruct_exemplar):
 	Call constructor with do_multi_index_op instead of feval.
 	Verify that symbol_table::find_method did return a class
 	constructor for the correct type.
 
 2011-01-26  John W. Eaton  <jwe@octave.org>
diff --git a/src/input.cc b/src/input.cc
--- a/src/input.cc
+++ b/src/input.cc
@@ -1250,27 +1250,27 @@ input_event_hook (void)
     {
       hook_fcn_map_type::iterator p = hook_fcn_map.begin ();
 
       while (p != hook_fcn_map.end ())
         {
           std::string hook_fcn = p->first;
           octave_value user_data = p->second;
 
-          p++;
+          hook_fcn_map_type::iterator q = p++;
 
           if (is_valid_function (hook_fcn))
             {
               if (user_data.is_defined ())
                 feval (hook_fcn, user_data, 0);
               else
                 feval (hook_fcn, octave_value_list (), 0);
             }
           else
-            hook_fcn_map.erase (p);
+            hook_fcn_map.erase (q);
         }
 
       if (hook_fcn_map.empty ())
         command_editor::remove_event_hook (input_event_hook);
     }
 
   return 0;
 }
